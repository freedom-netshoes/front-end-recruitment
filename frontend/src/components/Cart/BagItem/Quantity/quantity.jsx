import React, { Component } from 'react'
import { PropTypes } from 'prop-types'
import CSSModules from 'react-css-modules'

import styles from './quantity.less'

@CSSModules(styles)
export default class Quantity extends Component {
  static defaultProps = {
    quantity: 0
  };

  static propTypes = {
    quantity: PropTypes.number
  };

  render () {
    return (
      <p styleName='quantity'>Quantidade: {this.props.quantity}</p>
    )
  }
}
