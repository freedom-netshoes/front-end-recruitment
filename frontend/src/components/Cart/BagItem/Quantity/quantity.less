@import '../../../../styles/_less-variables.less';
@import '../../../../styles/_less-mixins.less';
.quantity {
    .fontLight();    
    color:@lightGray;
    font-size: 12px;
}