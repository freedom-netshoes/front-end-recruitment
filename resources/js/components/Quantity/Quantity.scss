@import '../../../scss/shared';

:global(#cart) {

  :global(.delete) .quantity {
    @include opacity(0.2);
    text-decoration: line-through;
  }

  .quantity {
    @include open-sans-light();
    @include transition();
    color    : $light-gray;
    font-size: 12px;
  }
}