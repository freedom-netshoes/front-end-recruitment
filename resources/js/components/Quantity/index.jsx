import React, {Component, PropTypes} from 'react';
import CSSModules from 'react-css-modules';
import styles from './Quantity';

@CSSModules(styles)
class Quantity extends Component {

  static defaultProps = {

  };

  static propTypes = {

  };

  render () {
    return (
      <p styleName='quantity'>Quantidade: 2</p>
    );
  }
}

export default Quantity;
