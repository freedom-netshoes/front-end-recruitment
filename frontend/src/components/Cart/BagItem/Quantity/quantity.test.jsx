/* global describe, it, expect */
import React from 'react'
import Quantity from './quantity'
import { shallow } from 'enzyme'

describe('Testing <Quantity /> component', () => {
  it('renders the Quantity component', () => {
    const component = shallow(<Quantity />)
    expect(component).toHaveLength(1)
  })

  it('testando snapshot do componente passando prop quantity', () => {
    let component = shallow(<Quantity quantity={12} />)
    expect(component).toMatchSnapshot()
  })
})
