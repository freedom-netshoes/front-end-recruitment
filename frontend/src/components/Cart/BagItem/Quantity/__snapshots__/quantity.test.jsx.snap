// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Quantity /> component testando snapshot do componente passando prop quantity 1`] = `
<p
  className="quantity"
>
  Quantidade: 
  12
</p>
`;
